### DISCUSSION TITLE
How do I reply to a message without it being sent as a draft?

### DISCUSSION THREAD

**Discussion by geoffda** [O365/python-o365#984] [Q&A] [answered]
I'm trying to reply to a message (code below), but it is always being sent as a draft. The message does get sent, but it shows up in the sent messages folder tagged as a draft, which I want to avoid. The message I'm replying to is not a draft (is_draft = False). However, from looking at the sources, messages always default to drafts and I can't figure out how to make them regular messages since there doesn't seem to be a way to set that property. In the graph api, the sendmail endpoint is the mechanism to not send as a draft, but in the o365 sources, message only uses that endpoint if __isdraft is False, which seems a bit circular. What am I missing to get my message tagged as not a draft? Thanks for any help.

```
    response_message = message.reply(False)
    response_message.to.add(referral_info.contact_email)
    response_message.to.remove(message.sender.address)
    response_message.body = "This is just a test at the moment."
    sent = response_message.send()
```

**Comment by alejcas** [comment]
I don't see them as drafts once they are send.

Check Ms graph docs to see alternatives to reply messages although I don't think there's any.

**Reply by geoffda** [reply]
Thanks for your reply. If you are able to call message.send and are getting a normal (non-draft) message in the sent folder, I wonder if is_draft == False in your case, and if so, how you got that set. AFIK, the send endpoint will send a draft, while the sendmessage endpoint will send a normal message. In the message.send implementation, the choice of endpoint is determined by the value of __isdraft. Could you possibly check the value of is_draft immediately prior to the message.send call? If you are able to debug the o365 library source, is there any chance you would be willing to step through message.send and check the value of __isdraft and confirm which endpoint is being used?

BTW, I don't normally develop in Python so if there is some easy way to enable source debugging of o365, please lmk and I'll be happy to do this myself. Anyway, since this is working for you, presumably your code is in a state to use the desired sendmail endpoint so the next question would be what are you doing differently to get the draft designation removed? Thanks again.

**Reply by geoffda** [reply]
Ah, never mind. I should have been looking at message.reply. Now I see that message.reply is using the createReply endpoint which creates a draft reply by design. The reply endpoint doesn't create a draft, but it doesn't appear to be implemented. So now I understand why my code is behaving the way that it is, but I still need to figure out how to just send a reply.

**Reply by geoffda** [reply, marked as answer]
Ok, figured out my problem. I was making changes to the message (which I now was a draft) without saving it. Apparently, this is what was causing the sent item to show up as a draft along with a few other weird behavioral things. Anyway, calling message.save_draft() prior to calling message.send() made everything behave as I was expecting.

**Reply by alejcas** [reply]
great you figure it!
